src/posix/platform/udp.cpp

```diff
@@ -402,9 +436,13 @@ void platformUdpInit(const char *aIfName)
         DieNow(OT_EXIT_INVALID_ARGUMENTS);
     }
 
-    sPlatNetifIndex = if_nametoindex(aIfName);
+    if (aIfName != gNetifName)
+    {
+        VerifyOrDie(strlen(aIfName) < sizeof(gNetifName) - 1, OT_EXIT_INVALID_ARGUMENTS);
+        strcpy(gNetifName, aIfName);
+    }
```

Comment:
Don't quite get the intention here. Any explanation? 

`gNetifIndex` is initialized by `platform/netif.cpp`, so it depends on `PLATFORM_NETIF`. 
But if `PLATFORM_NETIF` is `OFF`, how can we get `PLATFORM_UDP` work since it uses `gNetifIndex` in so many places.